tests/test_parameters.py

```diff
@@ -587,3 +587,15 @@ def test_compare_key_value_list():
 )
 def test_parameter_can_be_pickled(p):
     pickle.dumps(p)
+
+
+def test_add_remove_param():
```

Comment:
Can you explain in the test name what is being tested? Like what is adding and removing parameters supposed to do to pass the test? Just be executable, or is there some property they are supposed to maintain?